engine/src/main/java/com/datatorrent/stram/debug/TupleRecorder.java

```diff
@@ -463,10 +492,21 @@ private void publishTupleData(int portId, Object obj)
         wsClient.publish(recordingNameTopic, map);
       }
     } catch (Exception ex) {
-      logger.warn("Error publishing tuple data", ex);
+      if (shouldLog()) {
+        logger.warn("Error publishing tuple data {}", obj, ex);
```

Comment:
There will be no log entry if obj.toString() throws an exception. Assuming that the exception will be raised when an operator/TupleRecorder receives an unexpected tuple and it can't serialize it, it is very likely that obj.toString() will throw an exception as well.